From c29428b9bb0bdea4666dedef9c13c662f1ee17c0 Mon Sep 17 00:00:00 2001
From: Marcelo Leitner <mleitner@redhat.com>
Date: Tue, 21 Oct 2014 16:21:22 -0400
Subject: [net] netfilter: add explicit Kconfig for NETFILTER_XT_NAT

Message-id: <20141021162122.22888.93335.stgit@localhost.localdomain>
Patchwork-id: 98018
O-Subject: [RHEL7 net PATCH 121/124] netfilter: add explicit Kconfig for NETFILTER_XT_NAT
Bugzilla: 1148041 1155088
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>

Backport of upstream commit: 84a59ca55f699d1d1fbfffd75445bcfe0c3daf06

 commit 84a59ca55f699d1d1fbfffd75445bcfe0c3daf06
 Author: Pablo Neira Ayuso <pablo@netfilter.org>
 Date:   Thu Sep 4 09:47:21 2014 +0200

    netfilter: add explicit Kconfig for NETFILTER_XT_NAT

    Paul Bolle reports that 'select NETFILTER_XT_NAT' from the IPV4 and IPV6
    NAT tables becomes noop since there is no Kconfig switch for it. Add the
    Kconfig switch to resolve this problem.

    Fixes: 8993cf8 netfilter: move NAT Kconfig switches out of the iptables scope
    Reported-by: Paul Bolle <pebolle@tiscali.nl>
    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Marcelo Ricardo Leitner <mleitner@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/netfilter/Kconfig b/net/netfilter/Kconfig
index 349f233..6fc60a8 100644
--- a/net/netfilter/Kconfig
+++ b/net/netfilter/Kconfig
@@ -775,6 +775,14 @@ config NETFILTER_XT_TARGET_MARK
  (e.g. when running oldconfig). It selects
  CONFIG_NETFILTER_XT_MARK (combined mark/MARK module).
 
+config NETFILTER_XT_NAT
+ tristate '"SNAT and DNAT" targets support'
+ depends on NF_NAT
+ ---help---
+ This option enables the SNAT and DNAT targets.
+
+ To compile it as a module, choose M here. If unsure, say N.
+
 config NETFILTER_XT_TARGET_NETMAP
  tristate '"NETMAP" target support'
  depends on NF_NAT
-- 
1.7.1